litellm/llms/bedrock.py

```diff
@@ -282,6 +282,55 @@ def get_config(cls):
         }
 
 
+class AmazonMistralConfig:
+    """
+    Reference: https://docs.aws.amazon.com/bedrock/latest/userguide/model-parameters-mistral.html
+    Supported Params for the Amazon / Mistral models:
+
+    - `maxTokens` (integer) max tokens,
+    - `temperature` (float) temperature for model,
+    - `topP` (float) top p for model
+    - `stop` [string] A list of stop sequences that if generated by the model, stops the model from generating further output.
+    - `topK` (float) top k for model
+    """
+
+    maxTokens: Optional[int] = None
+    temperature: Optional[float] = None
+    topP: Optional[float] = None
+    topK: Optional[float] = None
+    stop: Optional[list[str]] = None
+
+    def __init__(
+        self,
+        maxTokens: Optional[int] = None,
```

Comment:
@ishaan-jaff do you guys have a preference for camelcase vs. underscore? pep8 should be max_tokens.

Reply:
hmm I noticed we're inconsistent in this file. Lets use camel case - since most of the other configs in this file are camel case

Reply:
@ishaan-jaff I tried camelCase and it will throw an error
```
litellm.exceptions.BadRequestError: BedrockException - An error occurred (ValidationException) when calling the InvokeModel operation: Malformed input request: #: extraneous key [topP] is not permitted, please reformat your input and try again.
```

The Bedrock docs show that they expect snake case for these models
https://docs.aws.amazon.com/bedrock/latest/userguide/model-parameters-mistral.html#model-parameters-mistral-request-response

Reply:
You need to translate the params here @ti3x 
https://github.com/BerriAI/litellm/blob/d494941dcceb27244a2434191bcc4da1c094f0ec/litellm/utils.py#L4486

Reply:
@ishaan-jaff got it, will take a look at it tomorrow.

Reply:
Happy to help out in any way, we'll make sure this is in prod by Monday for you

Reply:
@ishaan-jaff I added the mapping code in [a4e2476](https://github.com/BerriAI/litellm/pull/2290/commits/a4e24761a037624b11d4d565c3ad588e428d7286) so is there currently no way to map other non-openai optional params like top_k?
Works with 
```python
response = litellm.completion(
    model="bedrock/mistral.mistral-7b-instruct-v0:2",
    messages=messages,
    temperature=0,
    max_tokens=10,
    topP=0.7
)
```

I notice in the doc that a few places have             if (
                temperature == 0 and model == "mistral-7b-instruct"
            ):  # this model does no support temperature == 0
                temperature = 0.0001  # close to 0
            optional_params["temperature"] = temperature

Is this also required for Bedrock mistral-7b-instruct? Where is this 0 vs. 0.0001 documented?

Reply:
> temperature == 0 and model == "mistral-7b-instruct"
): # this model does no support temperature == 0
temperature = 0.0001
> Is this also required for Bedrock mistral-7b-instruct? Where is this 0 vs. 0.0001 documented?

I believe deepinfra/mistral had this bug. I'd recommend running with bedrock and seeing if you run into this bug

Reply:
> is there currently no way to map other non-openai optional params like top_k?

You can map that too, but map them in bedrock.py since they are bedrock specific

Reply:
@ishaan-jaff it looks good from what I can see. the top_k param is picked up by later code branches, so it's sending over correctly
```
Request Sent from LiteLLM:

                response = client.invoke_model_with_response_stream(
                    body={"prompt": "<s>[INST] Hello, how are you? [/INST]\n</s>", "max_tokens": 10, "temperature": 0, "top_p": 0.7, "top_k": 0.5},
                    modelId=mistral.mistral-7b-instruct-v0:2,
                    accept=accept,
                    contentType=contentType
                )
```

The temperature=0 has been what I am using for testing, and it did not throw any errors from Bedrock. I could not find any Mistral-specific docs about temp=0 being unsupported so it should be all good.